pkg/secrets/secrets_test.go

```diff
@@ -315,10 +315,12 @@ func TestSecretAWalk(t *testing.T) {
 	assert.Equal(t, "xx", secrets["keys_api_key"])
 	assert.Equal(t, "aaaa", secrets["credentials_api_a"])
 	assert.Equal(t, "bbbb", secrets["credentials_api_b"])
+	assert.Equal(t, "map[bar:baz]", secrets["credentials_api_foo"])
```

Comment:
Is there any benefit to capturing non-string types? I am not sure how often a consuming application might convert the string type to its actual type for use. I wonder if it's simpler to detect the type via `v.(type)` and only allow strings for now?

Reply:
If daytona should not handle other types, I would avoid panic by verifying the conversion. I agree this is probably not a frequent scenario.
However, I believe it should since in the README says:
```
vault login -token-only -method=$METHOD role=$VAULT_ROLE"
THING="$(vault read -field=key secret/path/to/thing)"
ANOTHER_THING="$(vault read -field=key secret/path/to/another/thing)"
echo $THING | app
...
```
So, that's why I tried to replicate this behavior.

Reply:
Technically it's never processed anything but strings or else it would have panic'd. I'm okay with clearly saying so in the README and doing a type assertion for strings only. Let me know what you think?

Reply:
In my use case I really would like reading dictionary since the key I want to load is a dictionary. Would using json.Marshal when we cannot convert to string work ?